### ISSUE TITLE
Difference between the routeKey in event object GET /projects/<id> on local vs GET /projects/{id} on AWS 

### ISSUE THREAD

**Issue by samaybhavsar** [aws/aws-sam-cli#3463]
### Description:

I created a project using SAM CLI v1.33.0 with Runtime: nodejs14.x. When running on my local machine using sam local start-api the routeKey in event object has angle brackets e.g. GET /projects/<id>  while after deploying on AWS the routeKey in event object has curly brackets e.g.  GET /projects/{id} . I am using the routeKey for access control.

Another similar issue which I noticed is, when running local I get "queryStringParameters": {}, (key with blank object) while after deploying to AWS, the key is not present.

### Steps to reproduce:

- Create a sample project using sam init
- Set up an API endpoint with custom field in URL. e.g. project/{project_id}
- Deploy the same to AWS
- Compare event object for both

### Observed result:

- routeKey has angle brackets instead of curly brackets
- "queryStringParameters": {}, on local but when deployed on AWS, this key is missing

### Expected result:
Should have curly brackets in routeKey

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 10
2. `sam --version`: 1.33.0
3. AWS region: ca-central-1

`Add --debug flag to command you are running`

**Comment by torresxb1** [comment]
Thanks for reporting the issue! I was able to reproduce in SAM CLI v1.35 for HttpApi with the following template:
```
Resources:
  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: hello-world/
      Handler: app.lambdaHandler
      Runtime: nodejs14.x
      Architectures:
        - x86_64
      Events:
        HelloWorld:
          Type: HttpApi 
          Properties:
            ApiId: !Ref MyApi
            Path: /project/{project_id}
            Method: get

  MyApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      StageName: Prod
```

In the handler, I just include the incoming event in the response and I do see `"routeKey":"GET /project/<project_id>"` and `"queryStringParameters":{}` when I run `sam local start-api` and curl the endpoint. When I curl the deployed endpoint, I see `"routeKey":"GET /project/{project_id}"` and `queryStringParameters` is not present. 

We've identified the issue, added it to our backlog, and will prioritize it accordingly. Thanks again for bringing this to our attention!

**Comment by mndeveci** [comment]
Released with [SAM CLI v1.38.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.38.0)
